lib/maths/analytics/CBoostedTreeFactory.cc

```diff
@@ -158,7 +178,8 @@ CBoostedTreeFactory::buildForTrain(core::CDataFrame& frame, std::size_t dependen
 
     this->startProgressMonitoringInitializeHyperparameters(frame);
 
-    if (this->initializeFeatureSampleDistribution()) {
+    if (m_TreeImpl->m_InitializationStage == CBoostedTreeImpl::E_EncodingInitialized ||
+        this->initializeFeatureSampleDistribution()) {
```

Comment:
On reflection, I don't think you can have this test here because we can failover at a later stage as part of `this->initializeHyperparameters(frame)`
```suggestion
    if (this->initializeFeatureSampleDistribution()) {
```

Reply:
Or perhaps better inside `CBoostedTreeFactory::initializeFeatureSampleDistribution()`:
```
    if (m_TreeImpl->m_FeatureSampleProbabilities.size() > 0) {
        return true;
    }
```